Add new languages and update translations for existing languages

AmitPhulera opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
https://dimagi.atlassian.net/browse/SAAS-17422

We are adding support for new languages - Arabic, Ukrainian and Italian and add the translations for remaining languages that we use on HQ.
## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
Generated via OpenAI GPT 4.1 using script [translate_po_files](https://github.com/dimagi/commcare-hq/pull/36335)
## Safety Assurance
Would put on staging to see deploy is smooth and languages are accessible on staging.
### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
NA
### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
NA


<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

AmitPhulera committed:
add langs arabic, italian, swahili and ukrainian in settings.py

also add the three letter langcodes in update_django_locales so that django is aware of these locales

AmitPhulera committed:
make directories for new langs by running mkdir -p locale/<lang_code>/LC_MESSAGES

after that run ./scripts/make-translations.sh

AmitPhulera committed:
run - ./manage.py translate_po_files --langs ar es fra hin ita sw ukr

gherceg commented on locale/en/LC_MESSAGES/djangojs.po:
```diff
@@ -914,6 +914,7 @@ msgid "Question"
 msgstr ""
 
 #: corehq/apps/app_manager/static/app_manager/js/summary/models.js
+#: corehq/apps/app_manager/static/app_manager/js/vellum/src/main-components.js
```
I assume the changes in this directory are simply because of changes since Wednesday that we haven't updated translations for yet? Just double checking my understanding 😄

AmitPhulera replied:
Yes. Your understanding is correct. These are the changes that happened on HQ code between Wednesday(when translations were merged) and yesterday when I ran make messages again.

gherceg reviewed (approved):
If my comment is wrong, dismiss my approval 😂 this looks good to me though. Is there a plan to verify these translations?

AmitPhulera committed:
rename arabic langcode from ar to ara - we use 3 digit lang code for arabic in HQ

AmitPhulera committed:
update settings files and add lang mapping in update_django_locales

AmitPhulera committed:
remove erroring translations

AmitPhulera committed:
add meta information to translation files to avoid warnings

locale/ukr/LC_MESSAGES/djangojs.po:8: warning: header field 'Language' still has the initial default value

AmitPhulera commented:
This has been deployed on staging and is working fine.

gherceg reviewed (commented):
Was cd604ad8e4becd39a7e73998c44b1f31d70c52ad a manual process? How do you intend to avoid translations that cause errors in the future?

AmitPhulera commented:
No. It was not a manual process. I ran `/manage.py translate_po_files --check-and-remove-errors` which it the utility that I have created recently.  This run `msgfmt` on the translation file and then removes all the errors that were flagged.
See the following functions - https://github.com/dimagi/commcare-hq/blob/b97e701fcb83bb4417b7436ea5875577c0128d14/corehq/apps/translations/management/commands/translate_po_files.py#L433-L535

gherceg commented:
> I ran /manage.py translate_po_files --check-and-remove-errors which it the utility that I have created recently

Ah so this is an additional command you need to run after initially generating translations? If there isn't an easy way to merge them into one command, are there clear docs for adding a new language to HQ so that if another dev needs to run through this process in the future, they can reference those?

AmitPhulera commented:
They are already merged in https://github.com/dimagi/commcare-hq/pull/36335/commits/ce89b5dea9464b344b35d64f785240d10fd789bb but they were not present at the time this PR was created. 😁 

There are docs in [this](https://docs.google.com/document/d/12-jRo_oeajzhwbhe7KV1tUwe10rvf7r_xRfCFDSvwNI/edit?tab=t.skqj4rc3cmg3) doc. Will put them in confluence after this PR is merged.
